pkg/kv/kvserver/consistency_queue.go

```diff
@@ -143,6 +145,14 @@ func consistencyQueueShouldQueueImpl(
 		if shouldQ, priority = shouldQueueAgain(now.ToTimestamp(), lpTS, data.interval); !shouldQ {
 			return false, 0
 		}
+
+		// Bump priority by up to +1 if the replica is for a range that has been
+		// recently modified.
+		// TODO(pavelkalinnikov): Consider bumping up to +K.
+		cliff := data.lastUpdateNanos + data.interval.Nanoseconds()
+		if n := now.ToTimestamp().GoTime().UnixNano(); n >= data.lastUpdateNanos && n < cliff {
+			priority += float64(cliff-n) / float64(data.interval.Nanoseconds())
```

Comment:
I think we probably want to extend this effect beyond a single `interval`, since clusters may not proceess all replicas within a single interval. Maybe over a week or so, possibly with exponential decay? We'll want the `lastUpdateNanos` to take precedence over the regular `interval` priority for this duration, so we may want to have a base priority of maybe 5 or 7 or something here.

Reply:
I think so too, was thinking about 3-5 here.

There are 2 load scenarios, depending on how long it takes to do a full scan:
1. Takes less than `interval`.
2. Takes up to `interval * M`, M >= 1.

For scenario (1), the priority scheme does not matter much. Most time the queue will be empty or have a few outstanding jobs, so the jobs can be slightly reordered depending on the scheme, but either way will complete within `interval`.

Scenario (2) is what we are targeting here. Without priority tweaks, every replica will be processed once per `interval * M`. With priority tweaks, it's somewhere between every `interval * max(1, M - spread)` and every `interval * (M + spread)`.

Trade-offs:

- If `spread` is too big, cold replicas are processed too rarely, but hot replicas tend to be processed every few `interval`s.
- If `spread` is low, we won't be able to process hot replicas once every `interval`, but all replicas get similar attention.

I think we should keep the `spread` relatively low. If it has to be big, then we are probably misconfigured, and the scan speed or `interval` should be adjusted.

Re exponential decay, I'm not too sure what the best function here would be, so used the simple linear for starters. Maybe the function should resemble the distribution in https://github.com/cockroachdb/cockroach/issues/87255#issuecomment-1277654504, but this one seems quite hyperbolic, or maybe [Zipfian](https://en.wikipedia.org/wiki/Zipf%27s_law). It's huge, and then it goes almost to zero and stays almost constant.

Reply:
Yeah, I think this makes sense. It's fairly intuitive and gives reasonable results. 4 seems like a reasonable value -- if a range hasn't seen writes for 4 days, it's  It would be worth spelling some of this out in a comment on `consistencyQueueShouldQueueImpl`.